front/app/api/ideas/useUpdateIdea.ts

```diff
@@ -43,6 +43,17 @@ const useUpdateIdea = () => {
       });
 
       const projectId = idea.data.relationships?.project.data.id;
+      const phases = idea.data.relationships?.phases;
+      const phaseIds = phases?.data?.map((phase) => phase.id);
+
+      // If the idea is a draft, we need to invalidate the cached draft idea
+      if (phaseIds && attributes.requestBody.publication_status === 'draft') {
+        phaseIds.forEach((phaseId) => {
+          queryClient.invalidateQueries({
+            queryKey: ideasKeys.item({ id: phaseId }),
+          });
+        });
+      }
```

Comment:
@EdwinKato I ended up finding another issue with caching of the draft survey responses. When we updated the idea we weren't invalidating the previous draft idea, so if we left the survey and came back without refreshing the page we were loading the _old_ draft response, not the latest one.

Assigning you to review again just in case this approach isn't correct :)